lib/controllers/github.js

```diff
@@ -0,0 +1,64 @@
+const { Router } = require('express');
+const jwt = require('jsonwebtoken');
+const authenticate = require('../middleware/authenticate');
+const GithubUser = require('../models/GithubUser');
+const {
+  getGithubProfile,
+  exchangeCodeForToken,
+} = require('../services/github');
+const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+module.exports = Router()
+  //send request to login page
+  .get('/login', async (req, res) => {
+    //use .env variable secrets as client_id and redirect uri made in github
+    res.redirect(
+      `https://github.com/login/oauth/authorize?client_id=${process.env.GITHUB_CLIENT_ID}&scope=user&redirect_uri=${process.env.GITHUB_REDIRECT_URI}`
+    );
```

Comment:
I believe the app is breaking here due to process.env.GITHUB_REDIRECT_URI being localhost instead of your deployed heroku site. Also, you may want to check your oauth app in github and ensure the /callback url matches the deployed site.